src/cursor/cur_stat.c

```diff
@@ -357,6 +357,7 @@ __curstat_conn_init(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst)
 	 * Optionally clear the connection statistics.
 	 */
 	__wt_conn_stat_init(session);
+	memset(&cst->u.conn_stats, 0, sizeof(WT_CONNECTION_STATS));
```

Comment:
Looking at this, I was thinking a slightly different change might be better:
```
diff --git a/src/cursor/cur_stat.c b/src/cursor/cur_stat.c
index 0118f3429..f4aee9a8c 100644
--- a/src/cursor/cur_stat.c
+++ b/src/cursor/cur_stat.c
@@ -357,6 +357,7 @@ __curstat_conn_init(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst)
         * Optionally clear the connection statistics.
         */
        __wt_conn_stat_init(session);
+       __wt_stat_connection_init_single(&cst->u.conn_stats);
        __wt_stat_connection_aggregate(conn->stats, &cst->u.conn_stats);
        if (F_ISSET(cst, WT_STAT_CLEAR))
                __wt_stat_connection_clear_all(conn->stats);
```
That is entirely untested and maybe I'm missing the point, so be careful.

Anyway, the idea is to make `__curstat_conn_init()` and `__curstat_file_init()` look the same.